pkg/util/cgroup/cgroup.go

```diff
@@ -0,0 +1,236 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+package cgroup
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"kubevirt.io/kubevirt/pkg/util"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/devices"
+	"github.com/opencontainers/runc/libcontainer/cgroups/ebpf"
+	"github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+
+	"golang.org/x/sys/unix"
+)
+
+var (
+	errNoPreciseMatch = errors.New("resulting devices cgroup doesn't precisely match target")
+	errNoMatch        = errors.New("resulting devices cgroup doesn't match target mode")
+)
+
+func BasePath() string {
+	return "/sys/fs/cgroup"
+}
+
+func ControllerPath(controller string) string {
+	if cgroups.IsCgroup2UnifiedMode() {
+		return BasePath()
+	}
+	return filepath.Join(BasePath(), controller)
+}
+
+func Slice(pid int, allowedControllers []string) (controllers []string, slice string, err error) {
+	cgroups, err := os.Open(fmt.Sprintf("/proc/%d/cgroup", pid))
```

Comment:
Maybe, like for the isolation detection, we can mock this file for unit testing. Now with cgroupv2, doing changes here becomes very delicate.

Reply:
Sure, will see how to make the funcion more unittest-friendly.